File: src/api/java/com/ultreon/mods/advanceddebug/api/common/Formatted.java
package com.ultreon.mods.advanceddebug.api.common;

import com.ultreon.mods.advanceddebug.api.client.formatter.IFormatterContext;

@SuppressWarnings("unused")
public final class Formatted implements IFormattable {
    private final String string;

    public String getString() {
        return string;
    }

    public Formatted(String string) {
        this.string = string;
    }

    public Formatted(Object object) {
        this.string = object.toString();
    }

    public Formatted(char c) {
        this.string = Character.toString(c);
    }

    public Formatted(byte b) {
        this.string = Byte.toString(b);
    }

    public Formatted(short s) {
        this.string = Short.toString(s);
    }

    public Formatted(int i) {
        this.string = Integer.toString(i);
    }

    public Formatted(long l) {
        this.string = Long.toString(l);
    }

    public Formatted(boolean b) {
        this.string = Boolean.toString(b);
    }

    @SuppressWarnings("deprecation")
    @Override
    public void format(IFormatterContext ctx) {
        ctx.direct(string);
    }
}

File: common/src/main/kotlin/com/ultreon/mods/advanceddebug/api/common/Formatted.kt
package com.ultreon.mods.advanceddebug.api.common

import com.ultreon.mods.advanceddebug.api.client.formatter.IFormatterContext

@Suppress("unused")
class Formatted : IFormattable {
    val string: String

    constructor(string: String) {
        this.string = string
    }

    constructor(`object`: Any?) {
        string = `object`.toString()
    }

    constructor(c: Char) {
        string = c.toString()
    }

    constructor(b: Byte) {
        string = b.toString()
    }

    constructor(s: Short) {
        string = s.toString()
    }

    constructor(i: Int) {
        string = i.toString()
    }

    constructor(l: Long) {
        string = l.toString()
    }

    constructor(b: Boolean) {
        string = java.lang.Boolean.toString(b)
    }

    @Suppress("DEPRECATION")
    override fun format(ctx: IFormatterContext) {
        ctx.direct(string)
    }
}

val Any?.formatted get() = Formatted(this)
val String.formatted get() = Formatted(this)
val Char.formatted get() = Formatted(this)
val Byte.formatted get() = Formatted(this)
val Short.formatted get() = Formatted(this)
val Int.formatted get() = Formatted(this)
val Long.formatted get() = Formatted(this)
val Float.formatted get() = Formatted(this)
val Double.formatted get() = Formatted(this)
val Boolean.formatted get() = Formatted(this)